### PR TITLE
docs: improve factories.rst

### PR DESCRIPTION
**Pull request by kenjis**
**Description**
- add another example
- add explanations

**Checklist:**
- [x] Securely signed commits
- [] Component(s) with PHPDoc blocks, only if necessary or adds value
- [] Unit testing, with >80% coverage
- [x] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

**Comment by kenjis** [comment]
@MGatner The following code loads `app/Libraries/Sub/SubLib.php`.
Is this a misuse or intended usage?
```php
$someLib = Factories::libraries('Sub/SubLib');
```

File: user_guide_src/source/concepts/factories.rst [hunk 1]
```diff
@@ -15,6 +15,11 @@ What are Factories?
 Like :doc:`./services`, **Factories** are an extension of autoloading that helps keep your code
 concise yet optimal, without having to pass around object instances between classes.
 
+Factories are similar to CodeIgniter 3's ``$this->load`` in the following points:
```
Comment (MGatner) [inline user_guide_src/source/concepts/factories.rst:18]: Is it worth adding CI3 references at this point? I don't know that the majority of our user base is coming from there, or recently enough to remember the CI3 mechanics.

**Review by michalsn — APPROVED** [review]

File: user_guide_src/source/concepts/factories.rst (hunk 1 above)
Comment (michalsn) [reply user_guide_src/source/concepts/factories.rst:18]: I see no harm. It can help some people to understand basic concepts.

**Comment by kenjis** [comment]
@MGatner See https://github.com/codeigniter4/CodeIgniter4/pull/7182#issuecomment-1404540767

**Comment by michalsn** [comment]
> @MGatner The following code loads `app/Libraries/Sub/SubLib.php`. Is this a misuse or intended usage?
> 
> ```
> $someLib = Factories::libraries('Sub/SubLib');
> ```

I hope it's intended. I use it a lot - not particularly with libraries.

**Comment by MGatner** [comment]
I don't recall intending that but I also don't have a problem with it!

**Review by MGatner — APPROVED** [review]

**Comment by kenjis** [comment]
Document it so that it is not removed as a bug. 
#7202

**Pull request merged**